visidata/clipboard.py

```diff
@@ -20,35 +24,76 @@
 globalCommand('gY', 'saveToClipboard(sheet, selectedRows or rows, input("copy rows to system clipboard as filetype: ", value=options.filetype or "csv"))', 'yank (copy) selected rows to system clipboard', 'data-clipboard-copy-system-selected')
 globalCommand('zY', 'copyToClipboard(cursorDisplay)', 'yank (copy) current cell to system clipboard', 'data-clipboard-copy-system-cell')
 
-option('clipboard_copy_cmd', 'pbcopy w', 'command to copy stdin to system clipboard')
-# or 'xsel --primary' for xsel
-# or 'xclip -selection primary'
-# or 'pbcopy w' for mac
+option('clipboard_copy_cmd', '', 'command to copy stdin to system clipboard')
 
-import tempfile
-import subprocess
+commands = {
+    ('clip', 'win32'):    [],                                      # Windows Vista+
+    ('pbcopy', 'darwin'): ['w'],                                   # macOS
+    ('xclip', None):      ['-selection', 'clipboard', '-filter'],  # Linux etc.
```

Comment:
If `-filter` is omitted, xclip runs as a daemon, which [blocks when run in a pipeline](https://github.com/asweigart/pyperclip/issues/116#issuecomment-387825778). This may not matter in VisiData (I haven't run it in a pipeline), but it shouldn't hurt. I think a better solution than launching xclip as a daemon every time would be to ensure the command's stdout is consumed and discarded (which is what I assumed was happening).

`clipboard` is the system clipboard targeted by <kbd>Ctrl-C</kbd>, <kbd>Ctrl-V</kbd>, and <kbd>Ctrl-X</kbd>. This is the standard clipboard used by browsers, text editors etc i.e. the apps most likely to be on the receiving end of pastes from VisiData. `primary` is the X clipboard accessible by clicking the left and middle mouse buttons. I believe that use case is less common (my guess is that most users are unaware of it), particularly for a keyboard-oriented tool, but anyone who has a preference for it can still, of course, configure it via the `clipboard_copy_cmd` option.

Reply:
Thanks for elaborating @chocolateboy and for the fix which discards the clipboard's stdout! Looks good.